docs/tutorials/entrypoints.rst

```diff
@@ -8,10 +8,96 @@
 Entry points
 ************
 
-
 .. todo::
 
-   - Mention what is the purpose of entry points
-   - Give an example of a console entry point
    - Give an example of a custom entry point
      - Mention pluggy maybe for an example use-case?
+
+Introduction
+============
+
+Entry points provide a mechanism to advertise components of an installed
+distribution to other code or users. Most notably, the ``console_scripts``
+entry points will create a command line wrapper.
+
+See the `PyPA documentation on entry points <https://packaging.python.org/en/latest/specifications/entry-points/>`_.
+
+Using entry points with meson-python
+====================================
+
+Entry points are defined in the ``pyproject.toml`` `metadata specification
+<https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#entry-points>`_.
+No further configuration is required when using ``meson-python``.
+
+Console entry point
+-------------------
+
+To show the usage of console entry points we build a simple
+python script:
+
+.. code-block:: python
+   :caption: src/simpleapp/console.py
+
+    """
+    Simple test application.
+
+    Usage:
+        simpleapp --help
+        simpleapp doc
+        simpleapp run [<file>] [options]
+
+    Options:
+        -h --help                                   display this help message
+        --workdir-path=<workdir-path>               directory in which to run [default: none]
+
+    """
+    import docopt
```

Comment:
It'd be better to use `argparse`, because it's in the stdlib - more well-known, and no separate install is needed to try this.